lib/arel/visitors/oracle_common.rb

```diff
@@ -12,6 +27,20 @@ def visit_Arel_Nodes_Matches(o, collector)
 
           super o, collector
         end
+
+        def cached_column_for(attr)
+          return unless Arel::Attributes::Attribute === attr
+
+          table = attr.relation.table_name
+          return unless schema_cache.columns_hash?(table)
+
+          column = attr.name.to_s
+          schema_cache.columns_hash(table)[column]
+        end
+
+        def schema_cache
+          @connection.schema_cache
+        end
```

Comment:
These two methods are mostly logic from [arel 7.1.4](https://github.com/rails/arel/blob/v7.1.4/lib/arel/visitors/to_sql.rb) with the difference that we only obtain cached columns. Otherwise we go into troubles trying to build the cache.